pkg/git/github/github_client.go

```diff
@@ -0,0 +1,402 @@
+/*
+Copyright 2024 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package github
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/google/go-github/v45/github"
+	"golang.org/x/oauth2"
+
+	gp "github.com/konflux-ci/mintmaker/pkg/git/gitprovider"
+)
+
+// Allow mocking for tests
+var NewGithubClient func(accessToken string) *GithubClient = newGithubClient
+var NewGithubClientWithBasicAuth func(username, password string) *GithubClient = newGithubClientWithBasicAuth
+
+const (
+	// Allowed values are 'json' and 'form' according to the doc: https://docs.github.com/en/rest/webhooks/repos#create-a-repository-webhook
+	webhookContentType = "json"
+)
+
+var (
+	appStudioPaCWebhookEvents = [...]string{"pull_request", "push", "issue_comment", "commit_comment"}
+)
+
+var _ gp.GitProviderClient = (*GithubClient)(nil)
+
+type GithubClient struct {
+	ctx    context.Context
+	client *github.Client
+
+	appId            int64
+	appPrivateKeyPem []byte
+}
+
+// EnsurePaCMergeRequest creates or updates existing Pipelines as Code configuration proposal merge request
```

Comment:
What is all this file about? Is this really on us?

Reply:
Most code in `pkg` are copied from build-service, I tried to remove some unrelated code segments, but there are still some code remains there, especially the `pkg/git` dir, I can continue on removing such code, or we can do that later. If I can complete it quickly, I'll update this MR.

Reply:
As discussed, we will handle the cleanup in future updates and will not address it immediately in this pull request.